ceremonies/src/lib.rs

```diff
@@ -531,39 +521,43 @@ impl<T: Config> Module<T> {
                         );
                         continue;
                     }
-                    let attestations = Self::attestation_registry(
+                    let attestees = Self::attestation_registry(
                         (cid, cindex),
                         &Self::attestation_index((cid, cindex), &p),
                     );
-                    if attestations.len() < (n_honest_participants - 1) as usize
-                        || attestations.is_empty()
-                    {
+                    if attestees.len() < (n_honest_participants - 1) as usize {
                         debug::debug!(
                             target: LOG,
-                            "skipped participant because of too few attestations ({}): {:?}",
-                            attestations.len(),
+                            "skipped participant because didn't testify for honest peers: {:?}",
                             p
                         );
                         continue;
                     }
-                    let mut has_attested = 0u32;
-                    for w in attestations {
-                        let w_attestations = Self::attestation_registry(
+
+                    let mut was_attested_count = 0u32;
+                    for other_participant in &meetup_participants {
```

Comment:
so, the price for improved UX is one more for loop over each meetup. At least it's a +- linear increase in complexity for meetups of <=12